packages/web/app/features/ProgramRegistry/PatientProgramRegistryUpdateFormModal.jsx

```diff
@@ -0,0 +1,337 @@
+  color: ${Colors.primary};
+  font-size: 0.6875rem;
+  text-decoration: underline;
+  padding: 0.3rem 1rem;
+  text-transform: none;
+  &:hover {
+    text-decoration: underline;
+  }
+`;
+
+const useUpdateProgramRegistryMutation = patientId => {
+  const api = useApi();
+  const queryClient = useQueryClient();
+
+  return useMutation(
+    data => {
+      return api.post(`patient/${patientId}/programRegistration`, data);
+    },
+    {
+      onSuccess: () => {
+        queryClient.invalidateQueries([`infoPaneListItem-${PANE_SECTION_IDS.PROGRAM_REGISTRY}`]);
+        queryClient.invalidateQueries(['patient', patientId]);
+      },
+    },
+  );
+};
+
+const getGroupedData = rows => {
+  const groupMapping = {
+    confirmedSection: [
+      'suspected',
+      'underInvestigation',
+      'confirmed',
+      'unknown',
+      'inRemission',
+      'notApplicable',
+    ],
+    resolvedSection: ['disproven', 'resolved'],
+    recordedInErrorSection: ['recordedInError'],
+  };
+
+  // Initialize result object
+  const groupedData = { confirmedSection: [{}], resolvedSection: [], recordedInErrorSection: [] };
+
+  // Process rows
+  rows.forEach(({ conditionCategory, date, programRegistryCondition }) => {
+    for (const [group, conditions] of Object.entries(groupMapping)) {
+      if (conditions.includes(conditionCategory)) {
+        groupedData[group].push({
+          conditionId: programRegistryCondition.id,
+          name: programRegistryCondition.name,
+          date,
+          conditionCategory,
+        });
+        break;
+      }
+    }
+  });
+  Object.keys(groupedData).forEach(group => {
+    groupedData[group].sort((a, b) => a.name?.localeCompare(b?.name));
+  });
+  return groupedData;
+};
+
+export const PatientProgramRegistryUpdateFormModal = ({
+  patientProgramRegistration = {},
+  onClose,
+  open,
+}) => {
+  const { programRegistryId, patientId, clinicalStatusId } = patientProgramRegistration;
+  const { mutateAsync: submit, isLoading: isSubmitting } = useUpdateProgramRegistryMutation(
+    patientId,
+  );
+  const { data: conditions = [], isLoading } = usePatientProgramRegistryConditionsQuery(
+    patientId,
+    programRegistryId,
+  );
+  const programRegistryStatusSuggester = useSuggester('programRegistryClinicalStatus', {
+    baseQueryParameters: { programRegistryId },
+  });
+
+  const handleSubmit = async data => {
+    await submit(data);
+    onClose();
+  };
+
+  if (!patientProgramRegistration) return null;
+
+  if (isLoading) {
+    return null;
+  }
+
+  return (
+    <Modal
+      title={
+        <TranslatedText
+          stringId="patientProgramRegistry.updateModal.title"
+          fallback="Update program registry"
+        />
+      }
+      open={open}
+      onClose={onClose}
+      width="lg"
+    >
+      <Form
+        showInlineErrorsOnly
+        onSubmit={handleSubmit}
+        render={({ setFieldValue, values, initialValues, dirty }) => {
+          const groupedData = values.conditions;
```

Comment:
The `values` field from Formik is the form state. I use this as the source of truth for the form values and also the layout of the form so that they automatically stay in sync. You will see `groupedData` (which needs a rename) set in the initialData of the form and passed into `StyledFormTable` below.